### ISSUE TITLE
Unexpected SecurityManager error on startup

### ISSUE THREAD

**Issue by grcevski** [elastic/elasticsearch#91658]
### Elasticsearch Version

8.x

### Java Version

_bundled_

### OS Version

Noticed on MacOS and Linux

### Problem Description

Sometimes when we encounter an error during node startup, depending on when the error happens we see the following SecurityManager error:

```
[2022-11-06T04:13:31,787][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [...] uncaught exception in thread [process reaper (pid 8927)]
java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "modifyThread")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:485) ~[?:?]
	at java.security.AccessController.checkPermission(AccessController.java:1068) ~[?:?]
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:411) ~[?:?]
	at org.elasticsearch.secure_sm.SecureSM.checkThreadAccess(SecureSM.java:166) ~[?:?]
	at org.elasticsearch.secure_sm.SecureSM.checkAccess(SecureSM.java:120) ~[?:?]
	at java.lang.Thread.checkAccess(Thread.java:2360) ~[?:?]
	at java.lang.Thread.setDaemon(Thread.java:2308) ~[?:?]
	at java.lang.ProcessHandleImpl.lambda$static$0(ProcessHandleImpl.java:103) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.<init>(ThreadPoolExecutor.java:637) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:928) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.processWorkerExit(ThreadPoolExecutor.java:1021) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1158) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1589) ~[?:?]
	at jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:186) ~[?:?]
```

Another instance I've seen:

```
[2022-11-08T09:32:33,417][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [elastic] uncaught exception in thread [process reaper (pid 83450)]java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "modifyThread")
	at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:485)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:1068)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:411)
	at org.elasticsearch.securesm@8.6.0-SNAPSHOT/org.elasticsearch.secure_sm.SecureSM.checkThreadAccess(SecureSM.java:166)
	at org.elasticsearch.securesm@8.6.0-SNAPSHOT/org.elasticsearch.secure_sm.SecureSM.checkAccess(SecureSM.java:120)

See logs for more details.
```

This usually happens when the error during bootstrap happens very early on, e.g:

```
[2022-11-06T04:13:29,508][INFO ][o.e.n.Node               ] [...] initialized
[2022-11-06T04:13:29,509][INFO ][o.e.n.Node               ] [...] starting ...
[2022-11-06T04:13:29,705][INFO ][o.e.x.s.c.f.PersistentCache] [...] persistent cache index loaded
[2022-11-06T04:13:29,707][INFO ][o.e.x.d.l.DeprecationIndexingComponent] [elksqa] deprecation component started
[2022-11-06T04:13:29,808][INFO ][o.e.t.TransportService   ] [elksqa] publish_address {...:9300}, bound_addresses {...:9300}
[2022-11-06T04:13:31,694][ERROR][o.e.b.Elasticsearch      ] [elksqa] fatal exception while booting Elasticsearch
```

The issue described here doesn't cause any problems, it's simply misleading for users when they are tracking an error in the logs and typically this is the last error reported.

### Steps to Reproduce

The only way to reproduce is to fake an error at the right time in startup, for example I've seen it happen when an error is hit inside `org.elasticsearch.gateway.PersistedClusterStateService.loadOnDiskState`.

**Comment by grcevski** [comment]
Duplicate of https://github.com/elastic/elasticsearch/issues/91650

**Comment by DaveCTurner** [comment]
> The issue described here doesn't cause any problems

A user approached me saying they were seeing nodes failing to start due to this issue, and I would have expected `uncaught exception in thread` to lead to process exit too. Is it really just log noise or is there something more to it?

**Comment by rendler-denis** [comment]
Hey, folks. I don't think this is a duplicate of #91704 since this crashes when trying to run the ES images mentioned in the documentation. At least, I have been trying for the last 2 days and using all 8.5.x tags, but I hit the same wall. I even removed apparmor from the system, but I still get this issue.

**Comment by rjernst** [comment]
> I would have `expected uncaught exception in thread` to lead to process exit

That error message is what we print for a non-fatal exception. A fatal exception (that will cause the node to exit) has the prefix `fatal error in thread`. So it should just be log noise.

> I don't think this is a duplicate of https://github.com/elastic/elasticsearch/pull/91704 since this crashes when trying to run the ES images mentioned in the documentation

This is a duplicate because the ultimate problem is the innocuous thread not having permissions. It surfaces in a couple different ways, and often hides other errors that would have occurred anyways. The issue is already fixed for 8.5.3 (yet to be released). If you are having ES fail startup, try looking for other exceptions in the log.
